antiplag/migrations/0015_auto_20210413_2157.py

```diff
@@ -0,0 +1,121 @@
+# Generated by Django 3.1.3 on 2021-04-13 19:57
+import uuid
+from uuid import UUID
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+from antiplag.encoders import UUIDEncoder
+
+
+def populate_total_percentage(apps, schema_editor):
+    """
+    Fill document's `total_percentage` attribute with results `percentage`
+    """
+    Result = apps.get_model("antiplag", "result")
+    Document = apps.get_model("antiplag", "document")
+
+    for result in Result.objects.all():
+        document = Document.objects.get(pk=getattr(result, "document_id"))
+        document.total_percentage = getattr(result, "percentage")
+        document.save()
+
+
+def migrate_result_to_new_model(apps, schema_editor):
+    Result = apps.get_model("antiplag", "result")
+    Document = apps.get_model("antiplag", "document")
+    Backup = apps.get_model("antiplag", "result_backup")
+
+    def parse_matched_doc(doc):
+        """
+        Parses each matched doc and returns dict of values after migrate
+        """
+        result = {}
+        try:
+            # if is valid UUID, it's internal Document id
+            result["match_id"] = str(UUID(doc["id"], version=5))
+            result["match_type"] = "UPLOADED"
+        except ValueError:
+            # else it's elastic id
+            result["match_id"] = doc["id"]
+            result["match_type"] = "CORPUS"
+
+        result["match_name"] = doc["name"]
+        result["ranges"] = doc["intervals"]
+        result["percentage"] = doc["percentage"]
+
+        return result
+
+    for result in Result.objects.all():
+        matched_docs = getattr(result, "matched_docs")
```

Comment:
Why are you not accessing `matches_docs` with dot notation?

Reply:
Not sure, I build upon kiko's function, therefore I used that. Will rework.

Reply:
Kiko's case was different. It was generic function, hence the use of `getattr`.